libmscore/barline.cpp

```diff
@@ -156,6 +156,12 @@ static void undoChangeBarLineType(BarLine* bl, BarLineType barType, bool allStav
     break;
     case BarLineType::START_REPEAT: {
         Measure* m2 = m->isMMRest() ? m->mmRestFirst() : m;
+        for (int staffIdx = 0; staffIdx < m2->score()->nstaves(); ++staffIdx) {
+            if (m2->isMeasureRepeatGroupWithPrevM(staffIdx)) {
+                MScore::setError(CANNOT_SPLIT_MEASURE_REPEAT);
```

Comment:
General comment: there are a ton of changes for MuseScore 4 that I have no insight into, so I'm just going to generally assume when I see something like this setError that I've never heard of before, that it's probably fine.